firestore-send-email/functions/src/helpers.ts

```diff
@@ -0,0 +1,21 @@
+import { createTransport } from "nodemailer";
+import { URL } from "url";
+import { Config } from "./types";
+
+export const setSmtpCredentials = (config: Config) => {
+  const { smtpConnectionUri, smtpPassword } = config;
+  let url = new URL(smtpConnectionUri);
+  let transport;
+
+  if (!url) {
```

Comment:
Can we log to the console here to say `invalid url, please reconfigure with a valid connection` and return early.